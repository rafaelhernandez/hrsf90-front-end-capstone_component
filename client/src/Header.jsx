import React from 'react';

const Header = (props) => (
  <div className="div-description-header">
    <div className="div-description-property-type">{props.desc.room_type_category === 'entire_home' ? ('Entire ' + props.desc.property_type).toUpperCase() : props.desc.property_type.toUpperCase()}</div>
    <div className="div-description-name">{props.desc.name}</div>
    <div className="div-description-city">{props.desc.city}</div>
    <div className="div-description-user-thumbnail_url">{props.desc.user.thumbnail_url}</div>
    <div className="div-description-guests">{props.desc.person_capacity === 1 ? '1 guest' : props.desc.person_capacity.toString() + ' guests'}</div>
    <div className="div-description-bedrooms">{props.desc.bedrooms === 0 ? 'Studio' : props.desc.bedrooms === 1 ? '1 Bedroom' : `${props.desc.bedrooms} Bedrooms`}</div>
    <div className="div-description-beds">{props.desc.beds === 1 ? '1 bed' : props.desc.beds.toString() + ' beds'}</div>
    <div className="div-description-bathrooms">{props.desc.bathrooms === 1 ? '1 bath' : props.desc.bathrooms.toString() + ' bathrooms'}</div>
  </div>
);

export default Header;
