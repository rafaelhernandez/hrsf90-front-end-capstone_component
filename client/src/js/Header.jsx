import React from 'react';

const Header = (props) => (
  <div className="div-description-header">
    <div className="div-description-property-type">{props.desc.room_type_category === 'entire_home' ? ('Entire ' + props.desc.property_type).toUpperCase() : props.desc.property_type.toUpperCase()}</div>
    <div className="div-description-name">{props.desc.name}</div>
    <div className="div-description-city">{props.desc.city}</div>
    <div className="div-description-user-thumbnail_url">{props.desc.user.thumbnail_url}</div>
    <div className="div-description-characteristics-block">
      <div className="div-description-characteristics-item">
        <div className="div-description-characteristics-table-cell">
          <i className="fas fa-users"></i>
        </div>
        <div className="div-description-characteristics-table-cell">
          <div className="div-description-guests">{props.desc.person_capacity === 1 ? '1 guest' : props.desc.person_capacity.toString() + ' guests'}</div>
        </div>
      </div>
    </div>
    <div className="div-description-characteristics-block">
      <div className="div-description-characteristics-item">
        <div className="div-description-characteristics-table-cell">
          <i className="fas fa-home"></i>
        </div>
        <div className="div-description-characteristics-table-cell">
          <div className="div-description-bedrooms">{props.desc.bedrooms === 0 ? 'Studio' : props.desc.bedrooms === 1 ? '1 Bedroom' : `${props.desc.bedrooms} Bedrooms`}</div>
        </div>
      </div>
    </div>
    <div className="div-description-characteristics-block">
      <div className="div-description-characteristics-item">
        <div className="div-description-characteristics-table-cell">
          <i className="fas fa-bed"></i>
        </div>
        <div className="div-description-characteristics-table-cell">
          <div className="div-description-beds">{props.desc.beds === 1 ? '1 bed' : props.desc.beds.toString() + ' beds'}</div>
        </div>
      </div>
    </div>
    <div className="div-description-characteristics-block">
      <div className="div-description-characteristics-item">
        <div className="div-description-characteristics-table-cell">
          <i className="fas fa-bath"></i>
        </div>
        <div className="div-description-characteristics-table-cell">
          <div className="div-description-bathrooms">{props.desc.bathrooms === 1 ? '1 bath' : props.desc.bathrooms.toString() + ' bathrooms'}</div>
        </div>
      </div>
    </div>
  </div>
);

export default Header;
