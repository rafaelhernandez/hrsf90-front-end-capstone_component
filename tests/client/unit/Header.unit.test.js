import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Header from '../../../client/src/Header.jsx';
import descriptionSample from '../../../db/data/description_sample.js';

configure({ adapter: new Adapter() });

describe('Header', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Header desc={descriptionSample}></Header>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
