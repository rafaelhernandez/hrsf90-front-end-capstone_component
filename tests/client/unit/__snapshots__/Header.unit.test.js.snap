// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  className="div-description-header"
>
  <div
    className="div-description-property-type"
  >
    ENTIRE HOUSE
  </div>
  <div
    className="div-description-name"
  >
    Garden Retreat steps from Haight St
  </div>
  <div
    className="div-description-city"
  >
    San Francisco
  </div>
  <div
    className="div-description-user-thumbnail_url"
  />
  <div
    className="div-description-characteristics-block"
  >
    <div
      className="div-description-characteristics-item"
    >
      <div
        className="div-description-characteristics-table-cell"
      >
        <i
          className="fas fa-users"
        />
      </div>
      <div
        className="div-description-characteristics-table-cell"
      >
        <div
          className="div-description-guests"
        >
          3 guests
        </div>
      </div>
    </div>
  </div>
  <div
    className="div-description-characteristics-block"
  >
    <div
      className="div-description-characteristics-item"
    >
      <div
        className="div-description-characteristics-table-cell"
      >
        <i
          className="fas fa-home"
        />
      </div>
      <div
        className="div-description-characteristics-table-cell"
      >
        <div
          className="div-description-bedrooms"
        >
          1 Bedroom
        </div>
      </div>
    </div>
  </div>
  <div
    className="div-description-characteristics-block"
  >
    <div
      className="div-description-characteristics-item"
    >
      <div
        className="div-description-characteristics-table-cell"
      >
        <i
          className="fas fa-bed"
        />
      </div>
      <div
        className="div-description-characteristics-table-cell"
      >
        <div
          className="div-description-beds"
        >
          2 beds
        </div>
      </div>
    </div>
  </div>
  <div
    className="div-description-characteristics-block"
  >
    <div
      className="div-description-characteristics-item"
    >
      <div
        className="div-description-characteristics-table-cell"
      >
        <i
          className="fas fa-bath"
        />
      </div>
      <div
        className="div-description-characteristics-table-cell"
      >
        <div
          className="div-description-bathrooms"
        >
          1 bath
        </div>
      </div>
    </div>
  </div>
</div>
`;
